### PR TITLE
Use subnet.dns_servers

### PR DESCRIPTION
**Pull request by ekohl**
Foreman 1.22 introduced the `subnet.dns_servers` helper which guarantees a
list of nameservers.

Requires https://github.com/theforeman/foreman/pull/6541

Please review carefully. I may have missed some things and we don't have tests for these.

### CODE REVIEW

File: provisioning_templates/snippet/kickstart_ifcfg_generic_interface.erb [hunk 1]
```diff
@@ -37,10 +37,9 @@ PEERDNS=<%= primary %>
 PEERROUTES=<%= primary %>
 DEFROUTE=<%= primary %>
 <%- if @interface.primary -%>
-<%-   if !@dhcp && @subnet && @subnet.dns_primary.present? -%>
-<%=     "DNS1=\"#{@subnet.dns_primary}\"" %>
-<%-     if @subnet.dns_secondary.present? -%>
-<%=       "DNS2=\"#{@subnet.dns_secondary}\"" %>
+<%-   if !@dhcp && @subnet -%>
+<%-     @subnet.dns_servers.each_with_index { |nameserver, index| -%>
```
Comment (ekohl) [inline provisioning_templates/snippet/kickstart_ifcfg_generic_interface.erb]: I don't know about the compatibility of `each_with_index`.

File: provisioning_templates/snippet/rancheros_cloudconfig.erb [hunk 2]
```diff
@@ -9,13 +9,10 @@ hostname: <%= @host.shortname %>
 rancher:
   network:
     dns:
+<% unless @host.subnet.dns_servers.empty? -%>
```
Comment (ekohl) [inline provisioning_templates/snippet/rancheros_cloudconfig.erb:12]: This is a change, but I think the previous code was incorrect. At least the `if ... or ...` statement in the previous version was redundant.

**Comment by ekohl** [comment]
Core has been merged. Needs a review now.

File: provisioning_templates/snippet/kickstart_ifcfg_generic_interface.erb (hunk 1 above)
Comment (lzap) [reply provisioning_templates/snippet/kickstart_ifcfg_generic_interface.erb]: It's not allowed in safemode, please avoid it.

File: provisioning_templates/snippet/rancheros_cloudconfig.erb (hunk 2 above)
Comment (lzap) [reply provisioning_templates/snippet/rancheros_cloudconfig.erb:12]: Yes.

**Review by lzap — CHANGES_REQUESTED** [review]
Safemode problem, other than that looks good.

**Comment by ekohl** [comment]
Any advice on how to avoid the safemode problem?

**Comment by lzap** [comment]
Just use `each` or `each_index` they are both allowed. I am working on adding it to safemode but I'd not rely on this.

**Comment by lzap** [comment]
FYI https://github.com/svenfuchs/safemode/pull/29

**Comment by ekohl** [comment]
Updated: rebased and now uses `each_index` instead of `each_with_index`

**Comment by lzap** [comment]
Thanks @ekohl merged.

**Pull request merged**